src/compute/src/sink/persist_sink.rs

```diff
@@ -720,11 +838,29 @@ where
                                     persist_frontier
                                 );
                             }
-                            correction.with_correction_buffer(sink_metrics, sink_worker_metrics, |buffer| buffer.append(&mut data));
+
+                            if let Some(stash) = &mut desired_stash {
+                                stash.insert(data);
```

Comment:
We might not have enough information here, but if we had the input's frontier, we could skip inserting the data in the stash if we know that it's due to be `drain_before`d immediately.

Reply:
Yep, I don't think the async operator lets us get that information.